ruby/lib/minitest/queue.rb

```diff
@@ -256,6 +259,25 @@ def run_from_queue(reporter, *)
           reporter.record(result)
         end
       end
+
+      queue.monitor.shutdown!
+    end
+
+    def with_heartbeat(example)
+      finished = false
+      t = Thread.start do
```

Comment:
```suggestion
      t = Thread.start do
        Thread.current.name = "CI::Queue#heartbeat"
```

Not strictly necessary, but it's nice to name threads.